internal/controller/helmchart_controller.go

```diff
@@ -1367,7 +1368,62 @@ func (r *HelmChartReconciler) makeVerifiers(ctx context.Context, obj *helmv1.Hel
 		}
 		verifiers = append(verifiers, verifier)
 		return verifiers, nil
+	case "notation":
+		// get the public keys from the given secret
+		secretRef := obj.Spec.Verify.SecretRef
+
+		if secretRef == nil {
+			return nil, fmt.Errorf("secretRef cannot be empty: '%s'", obj.Name)
+		}
+
+		pubSecret, _, err := r.retrieveSecret(ctx, obj.Namespace, secretRef.Name)
+		if err != nil {
+			return nil, err
+		}
+
+		data, ok := pubSecret.Data[defaultTrustPolicyKey]
+		if !ok {
+			return nil, fmt.Errorf("'%s' not found in secret '%s'", defaultTrustPolicyKey, secretRef.Name)
+		}
+
+		var doc trustpolicy.Document
+
+		if err := json.Unmarshal(data, &doc); err != nil {
+			return nil, err
+		}
+
+		defaultNotaryOciOpts := []soci.Options{
+			soci.WithTrustStore(&doc),
+			soci.WithNotaryRemoteOptions(verifyOpts...),
+		}
+
+		for k, data := range pubSecret.Data {
+			if strings.HasSuffix(k, ".crt") || strings.HasSuffix(k, ".pem") {
+				verifier, err := soci.NewNotaryVerifier(append(defaultNotaryOciOpts, soci.WithNotaryPublicCertificate(data), soci.WithNotaryAuth(clientOpts.Authenticator), soci.WithNotaryKeychain(clientOpts.Keychain), soci.WithInsecureRegistry(clientOpts.Insecure))...)
+				if err != nil {
+					return nil, err
+				}
+				verifiers = append(verifiers, verifier)
+			}
+		}
+		return verifiers, nil
 	default:
 		return nil, fmt.Errorf("unsupported verification provider: %s", obj.Spec.Verify.Provider)
 	}
 }
+
+// retrieveSecret retrieves a secret from the specified namespace with the given secret name.
+// It returns the retrieved secret and any error encountered during the retrieval process.
+func (r *HelmChartReconciler) retrieveSecret(ctx context.Context, ns string, secretName string) (corev1.Secret, string, error) {
```

Comment:
Resolved in 2da2a65

Reply:
My thinking was trying to reduce the need to double up on `verifySecret` in multiple places since it's only really needed to retrieve the secret and instead return the secret name for logging purposes. Happy either way.